src/Language/Haskell/TH/Instances.hs

```diff
@@ -140,6 +154,31 @@ instance Fail.MonadFail Q where
 # endif
 #endif
 
+#if !MIN_VERSION_template_haskell(2,17,0)
```

Comment:
Can you move this `#if !MIN_VERSION_template_haskell(2,17,0)` block of code closer to the end of this module, after the `#if MIN_VERSION_template_haskell(2,16,0)` block of code? Generally speaking, I like to order things by ascending order of the `template-haskell` version involved, since it makes it easier to search for. (I realize that this module is already a hornet's nest of CPP, but this makes it somewhat more tolerable.)

Reply:
Done